### ISSUE TITLE
Warnings with Block Comments—Spacing, and Extra Asterisks

### ISSUE THREAD

**Issue by theGeekPirate** [golang/lint#152]
If you begin a block comment with a space, golint does not recognize that it begins with the exported function name.

```
/* TestFunction does ... */
func TestFunction() {}
```

> comment on exported function TestFunction should be of the form "TestFunction ..."

The following passes without any warnings.

```
/*TestFunction does ...*/
```

Some people also prefer using multiple asterisks.

```
/***TestFunction does ...***/
```

> comment on exported function TestFunction should be of the form "TestFunction ..."

**Comment by mileusna** [comment]
+1

**Comment by kkirsche** [comment]
Does `gofmt` adjust block comments in any way which would dictate the "correctness" here?

**Comment by voigt** [comment]
Just stumbled upon this issue...
Is this really unintended? Where is this specified? Im about 3 days into go and not quite sure where to get the information about specs...

Just had a look into the source:

In my opinion [this](https://github.com/golang/go/blob/master/src/go/ast/ast.go#L109) switch-case is not working properly. There its defined what prefixes to strip from the comment (like `" "` or `"*"`...). I **think** it should also consider `"* "` (asterisk + whitespace).

```go
		// Remove comment markers.
		// The parser has given us exactly the comment text.
		switch c[1] {
		case '/':
			//-style comment (no newline at the end)
			c = c[2:]
			// strip first space - required for Example tests
			if len(c) > 0 && c[0] == ' ' {
				c = c[1:]
			}
		case '*':
			/*-style comment */
			c = c[2 : len(c)-2]
		}
```

Also I'm not 100% sure where else this function is used - this might have some unintended implications.

**Comment by dsnet** [comment]
I have an internal fix for this coming that will be exported soon.

**Comment by danesparza** [comment]
I like @voigt 's suggestion to strip asterisk or whitespace.  I format with a leading tab (*gasp*)  and the linter complains that my comment is formatted incorrectly

**Comment by josephholsten** [comment]
@dsnet did your fix for this get completed?

**Comment by Jesse-V** [comment]
@dsnet This issue appears to be outstanding in Golint 0.0+git20161013.3390df4-1, which is the latest version in Debian Stable (Stretch). It's a little annoying inside Atom; I look forward to the fix!

**Comment by andybons** [comment]
@dsnet what‘s the status on the internal fix?

**Comment by dsnet** [comment]
I apologize. I really dropped the ball on this one. I'm searching my email inbox and searched through all my CLs or CLs that I reviewed around that time frame, and I'm not sure what I was talking about.

**Comment by prkara** [comment]
+1 Extremely wierd bug..

**Comment by ndoell** [comment]
Just ran into this today.

**Comment by Jessicaward** [comment]
Yep, still there 😢

**Comment by jkalandaibm** [comment]
Had this 
Problem: 
/*test function bla bla bla*/
func CreateTestContext(....)
and was giving me the strange message: _warning: comment on exported function CreateTestContext should be of the form "CreateTestContext ..."_ 
**Solution:**
Ensure you include the name of the function in the comment itself, like so:
/*CreateTestContext bla bla bla*/ 

and that did the trick.
Even you have white-space between /* and function name, you'd still get a warning.
Pretty idiotic I must say, but for the time being, that solution works.

**Comment by freebiesoft** [comment]
is this project abandoned?

**Comment by kkirsche** [comment]
Thank you for the context, curious where was the proposal discussed? This seems to assume awareness of that initiative

**Comment by kkirsche** [comment]
My only feedback would be to link each reference to the deprecation thread to simplify jumping to the thread after completing reading. Currently the first link is the thread though the last paragraph also references it, though it wasn’t as clear that they were connected. Just my 2¢

**Comment by ianlancetaylor** [comment]
As the first line above indicates, this was discussed at  golang/go#38968.
